python/hail/api1/dataset.py

```diff
@@ -2363,7 +2364,27 @@ def grm(self):
         :rtype: :py:class:`KinshipMatrix`
         """
 
-        return KinshipMatrix(Env.hail().methods.GRM.apply(self._jvds))
+        print("Computing GRM...")
+
+        vds = (self.annotate_variants_expr(
+            'va.AC = gs.map(g => g.GT.gt).sum(), va.nCalled = gs.filter(g => isDefined(g.GT)).count()')
+               .filter_variants_expr('va.AC > 0 && va.AC < 2 * va.nCalled').persist())
+        n_variants = vds.count_variants()
+        if n_variants == 0:
+            raise FatalError("Cannot run GRM: found 0 variants after filtering out variants that are all homozygous reference or all homozygous variant.")
+        normalized_genotype_expr = '''let mean = va.AC / va.nCalled in
+            if (isDefined(g.GT)) (g.GT.gt - mean) / sqrt(mean * (2 - mean) * {} / 2)
+            else 0'''.format(n_variants)
+
+        f = new_temp_file(suffix="bm")
```

Comment:
Doesn't this use a local temp file, which doesn't fix the shuffle? We'll need to make sure this is getting a google storage temp file when running on the cloud.

Reply:
This pulls a temp file from wherever the user has defines as their temp directory (via input to hc). Maybe we just need to emphasize choosing a good temp directory?

Reply:
yeah, wasn't asking you to fix here! just need to make sure we keep this in mind going forward. Maybe we should read some config file for defaults and deploy sensible config with cloudtools